lib/db.js

```diff
@@ -302,6 +312,61 @@ class LimitDBRedis extends EventEmitter {
       });
   }
 
+  buildStatusResponse(bucket, prefixLength) {
+    return (key, lastDrip, remaining) => {
+      const bucketKeyConfig = this.bucketKeyConfig(bucket, key);
+      return {
+        key: key.substring(prefixLength),
+        limit: bucketKeyConfig.size,
+        reset: this.calculateReset(bucketKeyConfig, remaining),
+        remaining
+      };
+    };
+  }
+
+  /**
+   * Find all buckets that match the key.
+   *
+   * @param {statusParams} params
+   * @param {function(Error, statusResult)} [callback]
+   */
+  status(params, callback) {
+    const valError = this.validateParams(params);
+    if (valError) {
+      return setImmediate(callback, valError);
+    }
+    const bucket = this.buckets[params.type];
+    let prefix = `${params.type}:${params.key}*`;
+    let prefixLength = `${params.type}:`.length;
+
+    // Calling nodes directly does not use the global prefix
+    if (this.prefix) {
+      prefix = `${this.prefix}${prefix}`;
+      prefixLength = `${this.prefix}${params.type}:`.length
+    }
+
+    const count = params.count || 50;
+    const transform = this.buildStatusResponse(bucket, prefixLength);
+
+    const nodes = this.redis.nodes ? this.redis.nodes('slave') : [this.redis];
+    async.map(nodes, (node, cb) => {
+      node.bucketStatus(prefix, count, cb);
+    }, (err, results) => {
+      if (err) {
+        return callback(err);
+      }
+
+      const items = [];
+      for (let i = 0; i < results.length; i++) {
+        for (let j = 0; j < results[i].length; j++) {
```

Comment:
How much time does this holds the event loop assuming max number of items to return in all nodes?

Reply:
Proportional to the amount of items you requested and it finds. Hence the performance guarantees etc. This method is: "Use at your own risk".